src/components/multilineInput/SplitMultilineInput.tsx

```diff
@@ -0,0 +1,167 @@
+    defaultValues: ISplitValue[];
+    onChange?: (values: ISplitValue[]) => void;
+}
+
+export interface ISplitMultilineInputProps extends ISplitMultilineInputOwnProps {}
+
+export interface ISplitMultilineInputState {
+    values: ISplitValue[];
+}
+
+export class SplitMultilineInput extends React.PureComponent<ISplitMultilineInputProps, ISplitMultilineInputState> {
+
+    constructor(props: ISplitMultilineInputProps, state: ISplitMultilineInputState) {
+        super(props, state);
+        this.state = {
+            values: this.props.defaultValues,
+        };
+    }
+
+    componentDidUpdate(prevProps: ISplitMultilineInputProps) {
+        if (!_.isEqual(prevProps.defaultValues, this.props.defaultValues)) {
+            this.setState({
+                values: this.props.defaultValues,
+            });
+        }
+    }
+
+    render() {
+        return (
+            <div className={styles.inputBlock}>
+                <ul>
+                    {this.getDeletableInputs()}
+                    {this.getNewInput()}
+                </ul>
+            </div>
+        );
+    }
+
+    private getDeletableInputs() {
+        return _.map(this.state.values, (values: ISplitValue, index: number) => {
+            const labelIds: string[] = _.keys(values);
+            const inputLength: number = labelIds.length;
+            const inputs: JSX.Element[] = _.map(labelIds, (labelId: string, inputIndex: number) => {
+                const input: ISplitInput = _.findWhere(this.props.inputs, {id: labelId});
+
+                if (input) {
+                    const deleteButton: JSX.Element = inputIndex + 1 === inputLength ? <DeleteInputAction onClick={() => this.removeLine(index)} /> : null;
+                    return (
+                        <Input
+                            classes={styles.input}
+                            value={values[labelId]}
+                            placeholder={input.placeholder}
+                            validate={input.validation ? (value: any) => input.validation(value) : undefined}
+                            key={labelId + inputIndex}
+                            onChange={(value?: string, valid?: boolean) => this.changeValue(value, valid, index, labelId)}
+                        >
+                            <Label>{input.label}</Label>
+                            {deleteButton}
+                        </Input>
+                    );
+                }
+            });
+
+            return (
+                <li key={`split-${index}`} className={styles.inputLine}>
+                    {inputs}
+                </li>
+            );
+        });
+    }
+
+    private getNewInput() {
+        const inputRefs: Input[] = [];
+        const inputs: JSX.Element[] = _.map(this.props.inputs, (input: ISplitInput, inputIndex: number) => {
+            const addButton: JSX.Element = inputIndex + 1 === this.props.inputs.length ? <AddInputAction onClick={() => this.addLine(inputRefs)} /> : null;

```

Comment:
you could maybe add it inline with:  
   
```
{inputIndex === _.lastIndexOf(this.props.inputs)} 
    ? <AddInputAction onClick={() => this.addLine(inputRefs)} />
    : null}   
```  
you could remove the return and brackets too